manager/saw_manager.py

```diff
@@ -2,43 +2,154 @@
 from lab1python.models.chainsaw import Chainsaw
 from lab1python.models.circular_saw import CircularSaw
 from lab1python.models.jigsaw import Jigsaw
+import set_manager
 
-chainsaw1 = Chainsaw()
-electric_saw1 = ElectricSaw()
-circular_saw1 = CircularSaw()
-jigsaw1 = Jigsaw()
-chainsaw2 = Chainsaw(6.5, "T-1000", 1000, 4.1, 3.2,)
-electric_saw2 = ElectricSaw(7.1, "R2-D2", 1700, 80, True)
-circular_saw2 = CircularSaw(6.6, "C3-PO", 1500, 1.05, 0.03, True)
-jigsaw2 = Jigsaw(8, "HALO-10000", 550, 0.45, 0.03)
-list_of_saws = [chainsaw1, electric_saw1, circular_saw1, jigsaw1, chainsaw2, electric_saw2, circular_saw2, jigsaw2]
 
+def create_saws():
+    chainsaw1 = Chainsaw()
+    electric_saw1 = ElectricSaw()
+    circular_saw1 = CircularSaw()
+    jigsaw1 = Jigsaw()
+    chainsaw2 = Chainsaw(6.5, "T-1000", 1100, ("fuel", "electricity"), 4.1, 3.3, True)
+    electric_saw2 = ElectricSaw(7.1, "R2-D2", 1700, 80, ("battery", "electricity"), True)
+    circular_saw2 = CircularSaw(6.6, "C3-PO", 1500, 1.05, 0.03, ("battery", "electricity"))
+    jigsaw2 = Jigsaw(8, "HALO-10000", 550, 0.45, 0.03)
+    list_of_saws = [chainsaw1, electric_saw1, circular_saw1, jigsaw1, chainsaw2,
+                    electric_saw2, circular_saw2, jigsaw2]
+    return list_of_saws
 
+
+def numb_of_args(func):
+    def inner(*args):
+        number_of_args = len(args)
+        print(f"number of args = {number_of_args}")
+        return func(*args)
+    return inner
+
+
+def numb_of_calls(func):
+    counter = 0
+
+    def inner(*args):
+        nonlocal counter
+        counter += 1
+        if counter >= 3:
+            raise Exception("Too many calls")
+        return func(*args)
+    return inner
+
+
+@numb_of_calls
+def convert_saws_to_str():
+    string_saws = []
+    for i in create_saws():
+        string_saws.append(i.__str__())
```

Comment:
better not to use magic methods if you can use respective functions. For example, str(i) does exact same thing, but is readable